src/CycloneDX.Core/Models/Component.cs

```diff
@@ -197,14 +203,644 @@ public bool NonNullableModified
         public ReleaseNotes ReleaseNotes { get; set; }
         public bool ShouldSerializeReleaseNotes() { return ReleaseNotes != null; }
 
-        public bool Equals(Component obj)
+        public bool Equivalent(Component obj)
         {
-            return CycloneDX.Json.Serializer.Serialize(this) == CycloneDX.Json.Serializer.Serialize(obj);
+            return (!(obj is null) && this.BomRef == obj.BomRef);
```

Comment:
This is one example: in my opinion, equivalence should not just be based on the bomref.

Reply:
Fair point, I thought about tossing in the name/group/vendor/purl values, as they were checked in original (not mine) proposal for better flat merges in February: https://github.com/CycloneDX/cyclonedx-dotnet-library/pull/199/files#diff-0569878561c448041435a7c88dcd816279a2889bcc8e18193a2484fda5c15002R199-R205

But on one hand, what if those are not set (can check for that - skip if either is null, must be equal if not null)?

On another, at least as equivalence is checked now as part of `MergeWith()` as a quick filter, anyway these fields will be handled by the default case in the property loop, so if they are not equal - a `false` is returned and no merge happens. The loop deliberately uses a `Component tmp` clone of `this` just for that - to drop the attempt with no consequences. Only if all was ok the merged-or-unmodified values are transplanted back into `this`.